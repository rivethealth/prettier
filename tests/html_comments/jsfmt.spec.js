run_spec(__dirname, ["html"]);
run_spec(__dirname, ["html"], { printWidth: 1 });
run_spec(__dirname, ["html"], { printWidth: 999 });
