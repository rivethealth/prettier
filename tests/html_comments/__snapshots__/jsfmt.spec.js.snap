// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`before-text.html - html-verify 1`] = `
<!-- hello -->

123
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
<!-- hello -->

123

`;

exports[`before-text.html - html-verify 2`] = `
<!-- hello -->

123
~
<!--
  hello
-->

123

`;

exports[`before-text.html - html-verify 3`] = `
<!-- hello -->

123
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
<!-- hello -->

123

`;

exports[`conditional.html - html-verify 1`] = `
<!DOCTYPE html>
<html>
  <body>

    <!--[if IE 5]>This is IE 5<br><![endif]-->
    <!--[if IE 6]>This is IE 6<br><![endif]-->
    <!--[if IE 7]>This is IE 7<br><![endif]-->
    <!--[if IE 8]>This is IE 8<br><![endif]-->
    <!--[if IE 9]>This is IE 9<br><![endif]-->

  </body>
</html>
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
<!DOCTYPE html>
<html>
  <body>
    <!--[if IE 5]>This is IE 5<br /><![endif]-->
    <!--[if IE 6]>This is IE 6<br /><![endif]-->
    <!--[if IE 7]>This is IE 7<br /><![endif]-->
    <!--[if IE 8]>This is IE 8<br /><![endif]-->
    <!--[if IE 9]>This is IE 9<br /><![endif]-->
  </body>
</html>

`;

exports[`conditional.html - html-verify 2`] = `
<!DOCTYPE html>
<html>
  <body>

    <!--[if IE 5]>This is IE 5<br><![endif]-->
    <!--[if IE 6]>This is IE 6<br><![endif]-->
    <!--[if IE 7]>This is IE 7<br><![endif]-->
    <!--[if IE 8]>This is IE 8<br><![endif]-->
    <!--[if IE 9]>This is IE 9<br><![endif]-->

  </body>
</html>
~
<!DOCTYPE html>
<html>
  <body>
    <!--[if IE 5
      ]>This
      is
      IE
      5<br
    /><![endif]-->
    <!--[if IE 6
      ]>This
      is
      IE
      6<br
    /><![endif]-->
    <!--[if IE 7
      ]>This
      is
      IE
      7<br
    /><![endif]-->
    <!--[if IE 8
      ]>This
      is
      IE
      8<br
    /><![endif]-->
    <!--[if IE 9
      ]>This
      is
      IE
      9<br
    /><![endif]-->
  </body>
</html>

`;

exports[`conditional.html - html-verify 3`] = `
<!DOCTYPE html>
<html>
  <body>

    <!--[if IE 5]>This is IE 5<br><![endif]-->
    <!--[if IE 6]>This is IE 6<br><![endif]-->
    <!--[if IE 7]>This is IE 7<br><![endif]-->
    <!--[if IE 8]>This is IE 8<br><![endif]-->
    <!--[if IE 9]>This is IE 9<br><![endif]-->

  </body>
</html>
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
<!DOCTYPE html>
<html>
  <body>
    <!--[if IE 5]>This is IE 5<br /><![endif]-->
    <!--[if IE 6]>This is IE 6<br /><![endif]-->
    <!--[if IE 7]>This is IE 7<br /><![endif]-->
    <!--[if IE 8]>This is IE 8<br /><![endif]-->
    <!--[if IE 9]>This is IE 9<br /><![endif]-->
  </body>
</html>

`;

exports[`for_debugging.html - html-verify 1`] = `
<!DOCTYPE html>
<html>
  <body>

<!-- Do not display this at the moment
<img border="0" src="pic_trulli.jpg" alt="Trulli">
-->

  <!-- Do not display this at the moment
  <img border="0" src="pic_trulli.jpg" alt="Trulli">
  -->

    <!-- Do not display this at the moment
    <img border="0" src="pic_trulli.jpg" alt="Trulli">
    -->

  </body>
</html>
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
<!DOCTYPE html>
<html>
  <body>
    <!--
      Do not display this at the moment
      <img border="0" src="pic_trulli.jpg" alt="Trulli">
    -->

    <!--
      Do not display this at the moment
      <img border="0" src="pic_trulli.jpg" alt="Trulli">
    -->

    <!--
      Do not display this at the moment
      <img border="0" src="pic_trulli.jpg" alt="Trulli">
    -->
  </body>
</html>

`;

exports[`for_debugging.html - html-verify 2`] = `
<!DOCTYPE html>
<html>
  <body>

<!-- Do not display this at the moment
<img border="0" src="pic_trulli.jpg" alt="Trulli">
-->

  <!-- Do not display this at the moment
  <img border="0" src="pic_trulli.jpg" alt="Trulli">
  -->

    <!-- Do not display this at the moment
    <img border="0" src="pic_trulli.jpg" alt="Trulli">
    -->

  </body>
</html>
~
<!DOCTYPE html>
<html>
  <body>
    <!--
      Do not display this at the moment
      <img border="0" src="pic_trulli.jpg" alt="Trulli">
    -->

    <!--
      Do not display this at the moment
      <img border="0" src="pic_trulli.jpg" alt="Trulli">
    -->

    <!--
      Do not display this at the moment
      <img border="0" src="pic_trulli.jpg" alt="Trulli">
    -->
  </body>
</html>

`;

exports[`for_debugging.html - html-verify 3`] = `
<!DOCTYPE html>
<html>
  <body>

<!-- Do not display this at the moment
<img border="0" src="pic_trulli.jpg" alt="Trulli">
-->

  <!-- Do not display this at the moment
  <img border="0" src="pic_trulli.jpg" alt="Trulli">
  -->

    <!-- Do not display this at the moment
    <img border="0" src="pic_trulli.jpg" alt="Trulli">
    -->

  </body>
</html>
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
<!DOCTYPE html>
<html>
  <body>
    <!--
      Do not display this at the moment
      <img border="0" src="pic_trulli.jpg" alt="Trulli">
    -->

    <!--
      Do not display this at the moment
      <img border="0" src="pic_trulli.jpg" alt="Trulli">
    -->

    <!--
      Do not display this at the moment
      <img border="0" src="pic_trulli.jpg" alt="Trulli">
    -->
  </body>
</html>

`;

exports[`hidden.html - html-verify 1`] = `
<!DOCTYPE html>
<html>
    <body>

        <!--This is a comment-->
        <!-- This is a comment -->
        <!--  This is a comment  -->
        <!--   This   is   a   comment   -->
        <p>This is a paragraph.</p>
        <!-- Comments are not displayed in the browser -->

    </body>
</html>
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
<!DOCTYPE html>
<html>
  <body>
    <!-- This is a comment -->
    <!-- This is a comment -->
    <!-- This is a comment -->
    <!-- This   is   a   comment -->
    <p>This is a paragraph.</p>
    <!-- Comments are not displayed in the browser -->
  </body>
</html>

`;

exports[`hidden.html - html-verify 2`] = `
<!DOCTYPE html>
<html>
    <body>

        <!--This is a comment-->
        <!-- This is a comment -->
        <!--  This is a comment  -->
        <!--   This   is   a   comment   -->
        <p>This is a paragraph.</p>
        <!-- Comments are not displayed in the browser -->

    </body>
</html>
~
<!DOCTYPE html>
<html>
  <body>
    <!--
      This is a comment
    -->
    <!--
      This is a comment
    -->
    <!--
      This is a comment
    -->
    <!--
      This   is   a   comment
    -->
    <p>
      This
      is
      a
      paragraph.
    </p>
    <!--
      Comments are not displayed in the browser
    -->
  </body>
</html>

`;

exports[`hidden.html - html-verify 3`] = `
<!DOCTYPE html>
<html>
    <body>

        <!--This is a comment-->
        <!-- This is a comment -->
        <!--  This is a comment  -->
        <!--   This   is   a   comment   -->
        <p>This is a paragraph.</p>
        <!-- Comments are not displayed in the browser -->

    </body>
</html>
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
<!DOCTYPE html>
<html>
  <body>
    <!-- This is a comment -->
    <!-- This is a comment -->
    <!-- This is a comment -->
    <!-- This   is   a   comment -->
    <p>This is a paragraph.</p>
    <!-- Comments are not displayed in the browser -->
  </body>
</html>

`;

exports[`surrounding-empty-line.html - html-verify 1`] = `
<ul><!-- 123
--><li>First</li><!-- 123
456
   789
--><li>Second</li><!--


    123
       456
          789


--><li>Second</li><!--


           123
        456
    789


--></ul>
<span><!--
--><span>a</span><!--
--><span>b</span><!--
--></span>

<span><!-- 1
--><span>a</span><!-- 2
--><span>b</span><!-- 3
--></span>

<span><!--
1 --><span>a</span><!--
2 --><span>b</span><!--
3 --></span>
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
<ul>
  <!-- 123
  --><li>First</li
  ><!--
    123
    456
       789
  --><li>Second</li
  ><!--
    123
       456
          789
  --><li>Second</li
  ><!--
           123
        456
    789
  -->
</ul>
<span
  ><!--
  --><span>a</span
  ><!--
  --><span>b</span
  ><!--
--></span>

<span
  ><!-- 1
  --><span>a</span
  ><!-- 2
  --><span>b</span
  ><!-- 3
--></span>

<span
  ><!-- 1
  --><span>a</span
  ><!-- 2
  --><span>b</span
  ><!-- 3
--></span>

`;

exports[`surrounding-empty-line.html - html-verify 2`] = `
<ul><!-- 123
--><li>First</li><!-- 123
456
   789
--><li>Second</li><!--


    123
       456
          789


--><li>Second</li><!--


           123
        456
    789


--></ul>
<span><!--
--><span>a</span><!--
--><span>b</span><!--
--></span>

<span><!-- 1
--><span>a</span><!-- 2
--><span>b</span><!-- 3
--></span>

<span><!--
1 --><span>a</span><!--
2 --><span>b</span><!--
3 --></span>
~
<ul>
  <!--
    123
  --><li
    >First</li
  ><!--
    123
    456
       789
  --><li
    >Second</li
  ><!--
    123
       456
          789
  --><li
    >Second</li
  ><!--
           123
        456
    789
  -->
</ul>
<span
  ><!--
  --><span
    >a</span
  ><!--
  --><span
    >b</span
  ><!--
--></span>

<span
  ><!--
    1
  --><span
    >a</span
  ><!--
    2
  --><span
    >b</span
  ><!--
    3
--></span>

<span
  ><!--
    1
  --><span
    >a</span
  ><!--
    2
  --><span
    >b</span
  ><!--
    3
--></span>

`;

exports[`surrounding-empty-line.html - html-verify 3`] = `
<ul><!-- 123
--><li>First</li><!-- 123
456
   789
--><li>Second</li><!--


    123
       456
          789


--><li>Second</li><!--


           123
        456
    789


--></ul>
<span><!--
--><span>a</span><!--
--><span>b</span><!--
--></span>

<span><!-- 1
--><span>a</span><!-- 2
--><span>b</span><!-- 3
--></span>

<span><!--
1 --><span>a</span><!--
2 --><span>b</span><!--
3 --></span>
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
<ul>
  <!-- 123
  --><li>First</li
  ><!--
    123
    456
       789
  --><li>Second</li
  ><!--
    123
       456
          789
  --><li>Second</li
  ><!--
           123
        456
    789
  -->
</ul>
<span
  ><!--
  --><span>a</span
  ><!--
  --><span>b</span
  ><!--
--></span>

<span
  ><!-- 1
  --><span>a</span
  ><!-- 2
  --><span>b</span
  ><!-- 3
--></span>

<span
  ><!-- 1
  --><span>a</span
  ><!-- 2
  --><span>b</span
  ><!-- 3
--></span>

`;
